bootstrap/src/main/java/org/jboss/windup/bootstrap/Bootstrap.java

```diff
@@ -346,6 +348,40 @@ private void run(List<String> args)
         }
     }
 
+    private void startFurnace() throws InterruptedException, ExecutionException
+    {
+        Future<Furnace> future = furnace.startAsync();
+        future.get(); // use future.get() to wait until it is started
+        long startTime = System.currentTimeMillis();
+        long maxWait = 1000L * 30L; // only wait at most 30 seconds for addons to start
+        while (true)
+        {
+            long currentTime = System.currentTimeMillis();
+            long timeWaited = currentTime - startTime;
+
+            boolean allStarted = true;
+            for (Addon addon : furnace.getAddonRegistry().getAddons())
+            {
+                if (!addon.getStatus().isStarted())
+                {
+                    allStarted = false;
+                }
+            }
+            if (allStarted)
+                break;
+
+            if (timeWaited > maxWait)
+            {
+                System.err.println("WARN: Not all addons started!");
+                for (Addon addon : furnace.getAddonRegistry().getAddons())
+                {
+                    if (!addon.getStatus().isStarted())
+                        System.err.println("WARN: " + addon.getId() + " status: " + addon.getStatus());
+                }
+            }
```

Comment:
I think we should have a `break;` inside this `if` otherwise RHAMT will wait more than 30 secs for all the addons to start (maybe forever is not all the addons get to `STARTED` status)